sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -594,21 +595,28 @@ func cleanUp(containerdSocket string, namespace string, containerID string) erro
 
 // parseImageURISpecialRegions mimics the parsing in ecr.ParseImageURI but
 // constructs the canonical ECR references while skipping certain checks.
-// We only do this for special regions that are not yet supported by the aws-go-sdk.
+// We only do this for special regions that are not yet supported by the aws-go-sdk and for ECR fips endpoints.
 // Referenced source: https://github.com/awslabs/amazon-ecr-containerd-resolver/blob/a5058cf091f4fc573813a032db37a9820952f1f9/ecr/ref.go#L70-L71
 func parseImageURISpecialRegions(input string) (ecr.ECRSpec, error) {
 	ecrRefPrefixMapping := map[string]string{
 		"ap-southeast-5": "ecr.aws/arn:aws:ecr:ap-southeast-5:",
 		"eu-isoe-west-1": "ecr.aws/arn:aws-iso-e:ecr:eu-isoe-west-1:",
 	}
+	// https://docs.aws.amazon.com/general/latest/gr/ecr.html
+	fipsSupportedRegionMapping := map[string]string{
+		"us-east-1":     "",
+		"us-east-2":     "",
+		"us-west-1":     "",
+		"us-west-2":     "",
+		"us-gov-east-1": "",
+		"us-gov-west-1": "",
+	}
```

Comment:
Are you using a map for fast look ups? Since there aren't that many regions, maybe using a list would be OK. Alternatively, we could follow what others did here and use a map of booleans:

https://github.com/aws/amazon-ecs-agent/blob/ababaac1a77ac7f05f069859b246afeb2d99012b/agent/s3/factory/factory.go#L53

Reply:
I'm using this map like a set, since Go std lib does not provide a set type. That's https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/204/files#diff-e5dea204d3b274d07b5340673eeaa9b44e00cdf41d6fe9d25db80b4638ab9352R644

```
_, isFips := fipsSupportedRegionMapping[region]
```


I'll clarify that in both the comment and variable name itself (rename to fipsSupportedRegionSet)